blueprints/component-test/index.js

```diff
@@ -4,13 +4,30 @@ var path          = require('path');
 var testInfo      = require('../../lib/utilities/test-info');
 var stringUtil    = require('../../lib/utilities/string');
 var getPathOption = require('../../lib/utilities/get-component-path-option');
-
+var EOL           = require('os').EOL;
 
 module.exports = {
-  description: 'Generates a component unit test.',
+  description: 'Generates a component integration or unit test.',
+  
+  availableOptions: [
+    {
+      name: 'test-type',
+      type: ['integration', 'unit'],
+      default: 'integration',
+      aliases:[
+        { 'i': 'integration'},
+        { 'u': 'unit'},
+        { 'integration': 'integration' },
+        { 'unit': 'unit' }
+      ]
+    }
+  ],
 
   fileMapTokens: function() {
     return {
+      __testType__: function(options) {
+        return options.locals.testType || 'integration';
```

Comment:
I'd like to clean this up so it's not necessary, but there are situations where [addon developers could override locals](https://github.com/ember-cli/ember-cli/issues/4183), making testType unavailable.